include/deal.II/fe/fe_system.h

```diff
@@ -1081,6 +1097,54 @@ class FESystem : public FiniteElement<dim,spacedim>
   friend class FE_Enriched<dim,spacedim>;
 };
 
+//------------------------variadic template constructor------------------------
+
+#ifndef DOXYGEN
+namespace internal
+{
+  namespace FESystem
+  {
+    namespace
+    {
+      unsigned int count_nonzeros(const std::vector<unsigned int> &vec)
+      {
+        return std::count_if(vec.begin(), vec.end(), [](const unsigned int i)
+        {
+          return i > 0;
+        });
+      }
+    }
+  }
+}
+
+template<int dim, int spacedim>
+template <class... FESystems, typename>
+FESystem<dim,spacedim>::FESystem (FESystems &&... fe_systems):
+  FiniteElement<dim,spacedim>
+  (FETools::Compositing::multiply_dof_numbers<dim,spacedim>(std::forward<FESystems>(fe_systems)...),
+   FETools::Compositing::compute_restriction_is_additive_flags<dim,spacedim> (std::forward<FESystems>(fe_systems)...),
+   FETools::Compositing::compute_nonzero_components<dim,spacedim>(std::forward<FESystems>(fe_systems)...)),
+  base_elements(internal::FESystem::count_nonzeros({std::forward<FESystems>(fe_systems).second...}))
+{
+  std::vector<const FiniteElement<dim,spacedim>*> fes;
+  std::vector<unsigned int> multiplicities;
+
+  const auto extract
+    = [&fes, &multiplicities]
+      (const std::pair<std::unique_ptr<FiniteElement<dim, spacedim>>, unsigned int> &fe_system)
+  {
+    fes.push_back(fe_system.first.get());
+    multiplicities.push_back(fe_system.second);
+  };
+  [](...) {}((extract(std::forward<FESystems>(fe_systems)),0)...);
+
+  //now the vectors are in the wrong order so reverse
+  std::reverse(fes.begin(), fes.end());
+  std::reverse(multiplicities.begin(), multiplicities.end());
```

Comment:
Can you explain why they're in the wrong order?

Reply:
The compiler just evaluated the parameters in `(extract(std::forward<FESystems>(fe_systems)),0)...)` in opposite order. This is gone witch changing to the other idiom.